src/Analytics/Adapter/ActiveCampaign.php

```diff
@@ -50,17 +63,300 @@ public function getName(): string
+     * @param string $email
+     * @return bool|int
+     */
+    public function contactExists(string $email): bool|int
+    {
+        try {
+            $result = $this->call('GET', '/api/3/contacts', [], [
+                'email' => $email
+            ]);
+
+            $result = json_decode($result, true);
+
+            if ($result && $result['meta']['total'] > 0) {
+                return $result['contacts'][0]['id'];
+            } else {
+                return false;
+            }
+        } catch (\Exception $e) {
+            Console::error($e->getMessage());
+            return false;
+        }
+    }
+
+    /**
+     * Create a contact
+     * 
+     * @param string $email
+     * @param string $firstName
+     * @param string $lastName
+     * @param string $phone
+     * @return bool
+     */
+    public function createContact(string $email, string $firstName = '', string $lastName = '', string $phone = ''): bool
+    {
+        $body = ['contact' => [
+            'email' => $email,
+            'firstName' => $firstName,
+            'lastName' => $lastName,
+            'phone' => $phone
+        ]];
+
+        try {
+            $this->call('POST', '/api/3/contacts', [
+                'Content-Type' => 'application/json'
+            ], $body);
+            return true;
+        } catch (\Exception $e) {
+            Console::error($e->getMessage());
+            return false;
+        }
+    }
+
+    /**
+     * Update contact
+     * 
+     * @param string $contactId
+     * @param string $email
+     * @param string $firstName
+     * @param string $lastName
+     * @param string $phone
+     * 
+     * @return bool
+     */
+    public function updateContact(string $contactId, string $email, string $firstName = '', string $lastName = '', string $phone = ''): bool
+    {
+        $body = ['contact' => [
+            'email' => $email,
+            'firstName' => $firstName,
+            'lastName' => $lastName,
+            'phone' => $phone
+        ]];
+
+        try {
+            $this->call('PUT', '/api/3/contacts/'.$contactId, [
+                'Content-Type' => 'application/json'
+            ], $body);
+
+            return true;
+        } catch (\Exception $e) {
+            Console::error($e->getMessage());
+            return false;
+        }
+    }
+
+    /** 
+     * Delete a contact 
+     * 
+     * @param string $email
+     * @return bool
+     */
+    public function deleteContact($email): bool {
+        $contact = $this->contactExists($email);
+
+        if (!$contact) {
+            return false;
+        }
+
+        try {
+            $this->call('DELETE', '/api/3/contacts/'.$contact);
+            return true;
+        } catch (\Exception $e) {
+            Console::error($e->getMessage());
+            return false;
+        }
+    }
+
+    /**
+     * Account Exists
+     * 
+     * @param string $domain
+     * @return bool|int
+     */
+    public function accountExists($name): bool|int
```

Comment:
doc bloc and param is inconsistend? `domain` and `name`.

Reply:
Also missing type in PHP